### ISSUE TITLE
list comprehension with %%time - magic does not work in ipython 7.4.0 but still works in 7.3.0

### ISSUE THREAD

**Issue by noragen** [ipython/ipython#11674]
Example code from dask:

In 1: 
```from time import sleep
import random

def inc(x):
    sleep(0.200)
    return x + 1

def double(x):
    sleep(0.200)
    return x + 1
    
def add(x, y):
    sleep(0.200)
    return x + y
```

In 2:
```
%%time 
data = list(range(1,9))

out = []
for x in data:
    y = inc(x)
    z = double(y)
    out += [z]
    
# Tree Reduction
while len(out) > 1:
    print (len(out))
    out = [add(out[i], out[i + 1]) for i in range(0, len(out), 2)]

total = out[0]
```
With
```
(base) C:\>python -c "import IPython; print(IPython.sys_info())"
{'commit_hash': 'dd83a2163',
 'commit_source': 'installation',
 'default_encoding': 'utf-8',
 'ipython_path': 'C:\\Anaconda3\\lib\\site-packages\\IPython',
 'ipython_version': '6.5.0',
 'os_name': 'nt',
 'platform': 'Windows-10-10.0.17763-SP0',
 'sys_executable': 'C:\\Anaconda3\\python.exe',
 'sys_platform': 'win32',
 'sys_version': '3.7.3 (default, Mar 27 2019, 17:13:21) [MSC v.1915 64 bit '
                '(AMD64)]'}
```

Then the correct Result is:
```
8
4
2
Wall time: 4.61 s

In [3]:
```

But with newer version 7.4.0 it gives:

```
8\
---------------------------------------------------------------------------\
NameError                                 Traceback (most recent call last)\
<timed exec> in <module>

<timed exec> in <listcomp>(.0)

NameError: name 'out' is not defined
```

![grafik](https://user-images.githubusercontent.com/12185092/55638391-581e4d00-57c7-11e9-83bc-67ccd97e5dad.png "Error Message in V7.4.0")

**Comment by wanick** [comment]
problem in https://github.com/ipython/ipython/issues/11659

**Comment by cdeil** [comment]
This issue can be closed.

It was fixed in the ipython 7.5.0 release recently:
https://ipython.readthedocs.io/en/stable/whatsnew/version7.html#ipython-7-5-0

The example above works again (when adding the missing `from time import sleep`)
